File: backend/src/main/java/hu/ibc/ibcpartners/partner/dto/DiscountDto.java
package hu.ibc.ibcpartners.partner.dto;

import hu.ibc.ibcpartners.partner.entity.DiscountStatus;

import java.time.LocalDate;

public record DiscountDto(
        Long id,
        Long sellerId,
        Long buyerId,
        Long transactionId,
        String sellerName,
        String buyerName,
        String description,
        LocalDate fulfillmentDate,
        Long amount,
        Long discount,
        DiscountStatus status) {
}

File: frontend/src/app/features/dashboard/user/profile/discount/discount-dto.ts
import {CommissionStatus} from '../../../../../shared/commission-status';

export interface DiscountDto {
  id: number;
  transactionId: number;
  userId: number;
  userName: string;
  sellerName: string;
  buyerName: string;
  description: string | null;
  fulfillmentDate: string | null;
  amount: number;
  discount: number;
  status: CommissionStatus;
}
